stage #2.5/module #4 REST Basics - Spring Web/#6.REST API Tools for Testing, Design, and Development/swagger_for_designing_and_documenting_rest_api.md

```diff
@@ -0,0 +1,201 @@
+        return new ApiInfoBuilder().title("Application Rest API")
+                .description("News Application API")
+                .contact(new Contact("MJC Supervisor", "mjc/finalexam", "mjc@epam.com"))
+                .license("Apache 2.0")
+                .licenseUrl("http://www.apache.org/licenses/LICENSE-2.0.html")
+                .version("0.0.1-SNAPSHOT")
+                .build();
+    }
+}
+```
+Here, You have added metadata information about the REST API such as API name, author, website, license and so on. You have also instructed Swagger to only generate documentation for the components present in the mjc.newsapplication.controller package.
+
+Since we have enabled Swagger, let us see the documentation of our API endpoints done by Swagger. This is rendered through Swagger UI in the following link:
+
+http://localhost:8080//swagger-ui.html#/news-controller
+
+![](./media/swagger_default_documentation.png "Swagger default documentation")
+
+Here, Swagger has put together the following information:
+
+- Document metadata (API name, license, website, contact and so on)
+-All REST endpoints with default information it can infer from code. Note that endpoint descriptions are method names
+
+These are the default information. If you want, you can provide a detailed description and information about the endpoints and operations.
+
+## Documenting Rest Controller
+Now Let's document the REST controller of your application explicitly. Swagger provides several annotations to add documentation metadata that it pulls while generating the documentation.
+
+For each of the REST endpoint and its associated operations, You can provide ApiOperation and their various responses with ApiResponses annotations.
+
+```Java
+@RestController
+@RequestMapping("/api/v1/news")
+@Api(produces = "application/json", value = "Operations for creating, updating, retrieving and deleting news in the application")
+public class NewsController {
+
+    @Autowired
+    private NewsRepository newsRepository;
+
+    @PostMapping
+    @ApiOperation(value = "Create a piece of news", response = ResponseEntity.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully created a piece of news"),
+            @ApiResponse(code = 401, message = "You are not authorized to view the resource"),
+            @ApiResponse(code = 403, message = "Accessing the resource you were trying to reach is forbidden"),
+            @ApiResponse(code = 404, message = "The resource you were trying to reach is not found"),
+            @ApiResponse(code = 500, message = "Application failed to process the request")
+    }
+    )
+    public ResponseEntity<News> createNews(@RequestBody News news){
+       
+            News savedNews = newsRepository.save(news);
+            return new ResponseEntity<News>(savedNews, HttpStatus.OK);
+       
+    }
+
+    @GetMapping
+    @ApiOperation(value = "View all news", response = Iterable.class)
```

Comment:
Will it be better to use a more descriptive response definition instead of so generic interface, like 'Iterable'?
https://www.baeldung.com/java-swagger-set-list-response